resource/store.go

```diff
@@ -245,15 +245,48 @@ func (s *Store) ForceDelete(ctx context.Context, kind string, identifier Identif
 }
 
 // List lists all resources of kind in the provided namespace, with optional label filters.
-func (s *Store) List(ctx context.Context, kind string, namespace string, filters ...string) (ListObject, error) {
+// It will auto-list all pages in the response using the perPage limit for requests.
+// To list a single page of results, use ListPage.
+func (s *Store) List(ctx context.Context, kind string, namespace string, perPage int, filters ...string) (ListObject, error) {
```

Comment:
NIT – I _think_ having `perPage`, `namespace` and `filters` grouped into some kind of options struct would make for better UX. Incidentally it would also make the UX of listing non-namespaced resources better, since you won't need to pass an empty string to list – simply not specifying the ns in the options would take care of that.

Reply:
Yeah, I went back and forth on this one. I can change it to use an options struct specific to stores.